impl/src/main/java/org/jboss/weld/injection/InterceptionFactoryImpl.java

```diff
@@ -112,15 +118,41 @@ public T createInterceptedInstance(T instance) {
         }
         used = true;
 
-        if (annotatedType.getJavaClass().isInterface()) {
-            throw InterceptorLogger.LOG.interceptionFactoryNotOnInstance(annotatedType.getJavaClass().getCanonicalName());
+        AnnotatedType<T> originalAT = configurator != null ? configurator.complete() : annotatedType;
+        // this is the resulting AT we want to work with
+        AnnotatedType<T> resultingAnnotatedType = null;
+        // if it is an interface, we will have to 'merge' the AT with the AT of whatever user supplied as an instance
+        if (originalAT.getJavaClass().isInterface()) {
+            AnnotatedType<?> instanceAnnotatedType = beanManager.createAnnotatedType(instance.getClass());
```

Comment:
In the future it might be useful to have this extracted into a reusable unit of code ;-)

Reply:
Not sure I want to see this kind of "AT merging" anywhere else ;-)
But yea, I can create some util class and put it there.

Reply:
I tried this but in the end decided not to do that.
The code relies on a hierarchy of those two ATs, works only on interceptor bindings and ultimately needs to throw factory specific exception.
I don't see much re-usability there to justify it, at least for the time being.